schema/equals_test.go

```diff
@@ -60,86 +60,122 @@ func TestEquals(t *testing.T) {
 	// The "copy known fields" section of these function is to break if folks
 	// add new fields without fixing the Equals function and this test.
 	funcs := []interface{}{
-		func(x Schema) bool {
+		func(x *Schema) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y Schema
+			y := new(Schema)
 			y.Types = x.Types
 			return x.Equals(y) == reflect.DeepEqual(x, y)
 		},
-		func(x TypeDef) bool {
+		func(x *TypeDef) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y TypeDef
+			y := new(TypeDef)
 			y.Name = x.Name
 			y.Atom = x.Atom
 			return x.Equals(y) == reflect.DeepEqual(x, y)
 		},
-		func(x TypeRef) bool {
+		func(x *TypeRef) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y TypeRef
+			y := new(TypeRef)
 			y.NamedType = x.NamedType
 			y.Inlined = x.Inlined
 			return x.Equals(y) == reflect.DeepEqual(x, y)
 		},
-		func(x Atom) bool {
+		func(x *Atom) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y Atom
+			y := new(Atom)
 			y.Scalar = x.Scalar
 			y.List = x.List
 			y.Map = x.Map
 			return x.Equals(y) == reflect.DeepEqual(x, y)
 		},
-		func(x Map) bool {
+		func(x *Map) bool {
+			if x == nil {
+				// skip nil checks as the method can not be called on nil
+				return true
+			}
 			if !x.Equals(x) {
 				return false
 			}
-			var y Map
```

Comment:
I still don't think you have to change that, you can keep it as a variable, and then just pass the pointer in the function below?

Reply:
And you don't have to change the signature either?

Reply:
Or do you need to change the signature to avoid the copy?

Reply:
I think it might be fine to copy here though ...

Reply:
I do, at least because it causes a warning about copied locks.
But that's just my linter stuff.

Reply:
I changed the variable part but would prefer to keep the signatures.
Or is there a reason to copy here?